docs/content/guide/component-router.ngdoc

```diff
@@ -855,7 +855,7 @@ function HeroDetailComponent(heroService) {
   ...
   this.gotoHeroes = function() {
     var heroId = this.hero && this.hero.id;
-    this.$router.navigate(['HeroList']);
+    this.$router.navigate(['HeroList', {id: heroId}]);
```

Comment:
Actually, this is used (and explained) a couple of sections below. Here we should not use `heroId`.
If anything, we could remove the `var heroId = ...` line.